### DISCUSSION TITLE
[V5] Which hosting platform you recommend ? #GoInProduction

### DISCUSSION THREAD

**Discussion by Reptiluka** [adonisjs/core#1909] [Help] [answered]
Hi !
My website is ready to go in production, but I have problem with the most important part -> going online
In the past I used often a shared hosting, running in PHP with Apache from few hosting website.
It was simple, juste upload the file with FTP and the website is online.

Now, with node.js, I search a lot in the web, and it looks like difficult.
I looked a little bit Digital ocean (droplet/app platform), Google Cloud (compute engine, app engine) and few others.

But I dont want to do 10000 commands, and deal with server security and create the database manually and deal again with security and port and user and manual SSL certificate and and...

Do you have a SIMPLE web hosting to run node.js website ?
I just want to upload my code from my Git, few things / deploy command and thats it !
I know I may be able to do all config "like a real programmer you know" , but I don't want spend time to do that, I just want to code, I just want to create !

Need your help, thanks ! :)

**Comment by hariprasath-yadav** [comment]
Go with digital ocean ubuntu,
and install Virtualmin(with Nginx) and Node JS

**Comment by Banashek** [comment]
Firstly, I don't think a framework should necessarily have a primary recommendation for hosting unless a hosting provider is sponsoring the project or there is certain integration advantages.

To answer your question though: If you want simple deployment and maintenance, go with a PaaS like [Heroku](https://www.heroku.com/), [Render](https://render.com/), [DigitalOcean AppPlatform](https://www.digitalocean.com/products/app-platform/), [Elastic Beanstalk](https://aws.amazon.com/elasticbeanstalk/), [Azure App Service](https://azure.microsoft.com/en-us/services/app-service/web/), etc.

Any of these should work fine.

**Reply by Reptiluka** [reply]
Thank for the reply ! Yes I know, my question yes it's related with AdonisJs but it can be other project in node.js in general.

My final choice was DigitalOcean, which I tested their new AppPlatform, but it kind of limited, and should use Github (Bitbucket,Gitlab coming soon) and should pay a 15$/month if I want mysql server , plus 5$/month for a 512mb "kind of vps".

Instead, for the same price, I take a droplet from DigitalOcean with the double of ram (1gb) for the same price compared to App platform, and combine with the incredible https://cleavr.io/ which do all the config stuff (security, database, etc.)

So, I have the power of a VPS which I have all the control, combined with a great CI tool and server monitoring

**Comment by Reptiluka** [comment, marked as answer]
My final choice : 
DigitalOcean droplet , with https://cleavr.io/ for auto-configuration and deploy my project
